iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -34,12 +34,35 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
             return new DigitalTwinClient(iotHubConnectionString.HttpsEndpoint, sharedAccessKeyCredential, handlers);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DigitalTwinClient"/> class.</summary>
+        /// <param name="connectionString">The IoT hub's connection string.</param>
+        /// <param name="transportSettings">The optional settings to apply to the underlying http client.</param>
+        public static DigitalTwinClient CreateFromConnectionString(string connectionString, HttpTransportSettings transportSettings)
+        {
+            connectionString.ThrowIfNullOrWhiteSpace(nameof(connectionString));
+            transportSettings.ThrowIfNull(nameof(transportSettings));
+
+            var iotHubConnectionString = IotHubConnectionString.Parse(connectionString);
+            var sharedAccessKeyCredential = new SharedAccessKeyCredentials(connectionString);
+            return new DigitalTwinClient(iotHubConnectionString.HttpsEndpoint, sharedAccessKeyCredential, transportSettings);
+        }
+
         private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, params DelegatingHandler[] handlers)
         {
             _client = new IotHubGatewayServiceAPIs(uri, credentials, handlers);
             _protocolLayer = new DigitalTwin(_client);
         }
 
+        private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, HttpTransportSettings transportSettings)
+        {
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
nit: this also needs explanation -> who is disposing the http client? caller/ digital twin client/ us?